### PR TITLE
[IMP] pos_restaurant:test merging of table

### PR DESCRIPTION
**Pull request by SalukiMakingCode**
Tour steps:
implement the test for mergin table in PoS

check that table are merge
are unmerge when order finalize
unlink merge table in edit mode
table merge still after reloading page

tasks id : 367492

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_restaurant/static/tests/tours/pos_restaurant.js [hunk 1]
```diff
@@ -165,14 +171,130 @@ registry.category("web_tour.tours").add("pos_restaurant_sync_second_login", {
...
+        Dialog.confirm("Open session"),
+        FloorScreen.clickTable("5"),
+        ProductScreen.clickDisplayedProduct("Coca-Cola"),
+        ProductScreen.selectedOrderlineHas("Coca-Cola", "1.0"),
+        ProductScreen.clickOrderButton(),
+        ProductScreen.orderlinesHaveNoChange()
+    ].flat(),
+});
+
+function mergeTableHelpers() {
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/pos_restaurant.js:186]: you should also add a docstring that explains that table 4 will be the parent of table 5

File: addons/pos_restaurant/static/tests/tours/pos_restaurant.js [hunk 2]
```diff
@@ -165,14 +171,130 @@ registry.category("web_tour.tours").add("pos_restaurant_sync_second_login", {
...
+            trigger: 'div.isLinked div.label:contains("4")',
+        },
+        {
+            content: `unlink in edit plan if unlink possible`,
+            trigger: '.edit-buttons button:contains("Unlink")',
+        },
+        {
+            content: `click on the burger menu`,
+            trigger: 'i.fa-bars',
+        },
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/pos_restaurant.js:272]: you can use the `clickMenuButton` function from `point_of_sale/static/tests/tours/helpers/ChromeTourMethods.js`

File: addons/pos_restaurant/static/tests/tours/pos_restaurant.js [hunk 3]
```diff
@@ -165,14 +171,130 @@ registry.category("web_tour.tours").add("pos_restaurant_sync_second_login", {
...
+        ...checkMergeTableIsCancelHelpers(),
+        ...mergeTableHelpers(),
+        {
+            content: `refresh page`,
+            trigger: 'div.table div.label:contains("4")',
+            isCheck: true,
+            run: () => {
+                $(document).ready(function () {
+                    location.reload();
+                });
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/pos_restaurant.js:286]: do you need the `$(document).ready` part? The page should be loaded at this point so i would think that you could simply do `run: () => window.location.reload()`

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_restaurant/tests/test_frontend.py [hunk 4]
```diff
@@ -14,7 +14,8 @@ class TestFrontend(AccountTestInvoicingCommon, HttpCaseWithUserDemo):
     @classmethod
     def setUpClass(cls, chart_template_ref=None):
         super().setUpClass(chart_template_ref=chart_template_ref)
-        cls.user_demo.groups_id += cls.env.ref('point_of_sale.group_pos_manager') + cls.env.ref('account.group_account_invoice')
+        cls.user_demo.groups_id += cls.env.ref('point_of_sale.group_pos_manager') + cls.env.ref(
+            'account.group_account_invoice')
```
Comment (vlst-odoo) [inline addons/pos_restaurant/tests/test_frontend.py:19]: please restore all this diff

File: addons/pos_restaurant/tests/test_frontend.py [hunk 5]
```diff
@@ -266,6 +267,14 @@ def test_09_combo_split_bill(self):
...
+        self.start_tour("/pos/ui?config_id=%d" % self.pos_config.id, 'SaveLastPreparationChangesTour',
+                        login="pos_admin")
+        self.assertTrue(self.pos_config.current_session_id.order_ids.last_order_preparation_change,
+                        "There should be a last order preparation change")
+        self.assertTrue("Coca" in self.pos_config.current_session_id.order_ids.last_order_preparation_change,
+                        "The last order preparation change should contain 'Coca'")
+
+    def test_11_merge_table(self):
+        self.pos_config.with_user(self.pos_admin).open_ui()
+        self.start_tour("/pos/ui?config_id=%d" % self.pos_config.id, 'MergeTableTour', login="pos_admin", watch=False,
```
Comment (vlst-odoo) [inline addons/pos_restaurant/tests/test_frontend.py:279]: ideally, this test should be run with `login="pos_user"`

**Comment by caburj** [comment]
Careful in merging. The commit doesn't have the correct email and name of the author.

File: addons/pos_restaurant/tests/test_frontend.py (hunk 5 above)
Comment (SalukiMakingCode) [reply addons/pos_restaurant/tests/test_frontend.py:279]: Don't get it, all the other test is with pos_admin and when I try with pos_user i have a : "odoo.exceptions.AccessDenied: Access Denied"

File: addons/pos_restaurant/static/tests/tours/pos_restaurant.js [hunk 6]
```diff
@@ -156,14 +158,124 @@ registry.category("web_tour.tours").add("pos_restaurant_sync_second_login", {
...
             Dialog.confirm("Open session"),
             FloorScreen.clickTable("5"),
             ProductScreen.clickDisplayedProduct("Coca-Cola"),
             ProductScreen.selectedOrderlineHas("Coca-Cola", "1.0"),
             ProductScreen.clickOrderButton(),
-            ProductScreen.orderlinesHaveNoChange()
+            ProductScreen.orderlinesHaveNoChange(),
         ].flat(),
-    });
+});
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/pos_restaurant.js:172]: please restore this diff

File: addons/pos_restaurant/tests/test_frontend.py [hunk 7]
```diff
@@ -228,7 +229,8 @@ def test_04_ticket_screen(self):
         self.start_tour("/pos/ui?config_id=%d" % self.pos_config.id, 'PosResTicketScreenTour', login="pos_admin")
 
     def test_05_tip_screen(self):
-        self.pos_config.write({'set_tip_after_payment': True, 'iface_tipproduct': True, 'tip_product_id': self.env.ref('point_of_sale.product_product_tip')})
```
Comment (vlst-odoo) [inline addons/pos_restaurant/tests/test_frontend.py:231]: please restore all unneeded diff

File: addons/pos_restaurant/tests/test_frontend.py [hunk 8]
```diff
@@ -266,6 +267,14 @@ def test_09_combo_split_bill(self):
...
+                        login="pos_admin")
+        self.assertTrue(self.pos_config.current_session_id.order_ids.last_order_preparation_change,
+                        "There should be a last order preparation change")
+        self.assertTrue("Coca" in self.pos_config.current_session_id.order_ids.last_order_preparation_change,
+                        "The last order preparation change should contain 'Coca'")
+
+    def test_11_merge_table(self):
+        self.pos_config.with_user(self.pos_admin).open_ui()
+        self.start_tour("/pos/ui?config_id=%d" % self.pos_config.id, 'MergeTableTour', login="pos_admin", watch=False,
+                        step_delay=1000)
```
Comment (vlst-odoo) [inline addons/pos_restaurant/tests/test_frontend.py:280]: watch and step_delay have to be removed before merging

File: addons/pos_restaurant/static/tests/tours/pos_restaurant.js [hunk 9]
```diff
@@ -156,14 +158,124 @@ registry.category("web_tour.tours").add("pos_restaurant_sync_second_login", {
...
+            trigger: 'i[aria-label="Merge"]',
+        },
+        FloorScreen.clickTable(parentName),
+        FloorScreen.backToFloor(),
+        {
+            content: `Verify table ${childName} is merged into table ${parentName}`,
+            trigger: `div.table div.label:contains("${parentName}")`,
+            isCheck: true,
+            run: () => {
+                if ($(`div.table div.label:contains('${parentName}')`).length < 2) {
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/pos_restaurant.js:189]: check that there are exactly 2 tables, not 2 or more

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**